ui/src/app/store/machine/utils.ts

```diff
@@ -0,0 +1,69 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { general as generalActions } from "app/base/actions";
+import generalSelectors from "app/store/general/selectors";
+import type { Machine } from "app/store/machine/types";
+
+/**
+ * Check if a machine has an invalid architecture.
+ * @param machine - A machine object.
+ * @returns Whether the machine has an invalid architecture.
+ */
+export const useHasInvalidArchitecture = (
+  machine?: Machine | null
```

Comment:
Can this also be `MachineDetail`?

Reply:
Yep! The `Machine` is `type Machine = BaseMachine | MachineDetails;`